### PR TITLE
Dev 31 refactor shared pref

### PR DESCRIPTION
**Pull request by yabdelkawy**
- This branch features refactored code, primarily in the Directions functionality. 
 
Now, the code uses the ViewModel architecture exclusively to obtain a user's origin and destination. 

Please ensure that 

- [ ] Switch button functionality works

- [ ] Alt directions functionality works

- [ ] Different transportation modes work 

- [ ] You can change the origin location

- [ ] You can change the destination location

### CODE REVIEW

**Comment by Khal-Shah** [comment]
This code is so much cleaner!

My observations: 

So for emulator, the initial "My Location" is shown as just the text "My Location" instead of an address. The actual initial location address seems to be hardcoded. This will be fine if we're scaling down a bit.

For phone: While trying to use the the 2nd commented out line, I'm getting "Unresolved Reference" error for getCurrentLocation in Line 48 of DirectionsFragment:
directionsViewModel.originLocation.value= Map.getCurrentLocation()
Error persists even after importing Map.getCurrentLocation().

Apart from that, everything seems to be functional and working properly. Thank you for your hard work!

**Comment by nguyhh** [comment]
Hmm when I run it on my phone the current location detected is wrong...

File: app/src/main/java/com/soen390/conumap/ui/directions/DirectionsFragment.kt [hunk 1]
```diff
@@ -1,69 +1,61 @@
...
+            this.viewmodel=path
         }
 
-        getPath()
 
-        val sharedPreferences: SharedPreferences =requireContext().getSharedPreferences("SearchCount",0)
+        //Initial transportation mode is already driving
+        if (directionsViewModel.destinationChanged.value==null)
+        {
+            //If you're using an emulator, use the first commented out line
```
Comment (nguyhh) [inline app/src/main/java/com/soen390/conumap/ui/directions/DirectionsFragment.kt:44]: Oh I just saw your comment here. I dont think we should hardcode the current location. Even on the emulator you can set your current location to whereever you want to.

Comment (yabdelkawy) [reply app/src/main/java/com/soen390/conumap/ui/directions/DirectionsFragment.kt:44]: the thing is, the initial location for me is in California so I hard coded the location to somewhere closer to Montreal so I don't get memory overflow errors. 
I still kept the current location code to test. I can take it out and we can see with just that code

**Comment by yabdelkawy** [comment]
@Khal-Shah so concerning the "My Location", that text is only displayed if a user doesn't modify their location as their location is the initial location as in the mockups. 
Now about the hardcoded location, @nguyhh @Khal-Shah, I fixed that with this new commit and it should work for the both of you on your phones and emulators. Please let me know if something is wrong! 

Thank you guys!

**Comment by nguyhh** [comment]
Good job! I am approving this please merge asap so we can get the other PR approved too.

**Comment by yabdelkawy** [comment]
@nguyhh I re-requested your review because it wasn't approved and I think you closed the PR before I could merge so it didn't merge with submaster

**Comment by nguyhh** [comment]
Ohh woops soo sorryy!!

**Pull request merged**